src/helpers.py

```diff
@@ -0,0 +1,25 @@
+#!/usr/bin/env python3
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Module containing helpers for the charm module."""
+
+import re
+
+
+def invalid_hostname_check(hostname: str) -> bool:
+    """Check if the hostname is valid according to RFC 1123.
+
+    Args:
+        hostname: Ingress hostname
+    """
```

Comment:
I'm surprised this passed since the docstring doesn't provide a description of the return value?

Reply:
This should have been caught by `flake8-docstrings-complete`

Reply:
> This should have been caught by `flake8-docstrings-complete`

Perhaps we can write a test for this in `operator-workflows`?

Reply:
Perhaps more for the linter, I'll check what happened

Reply:
it turns out the problem is that this charm has a `.flake8` file that enables only a handful of checks. We should probable create a ticket to remove that and fix all the linting issues, there are a lot of problems with documentation for this charm

```
./lib/charms/nginx_ingress_integrator/v0/ingress.py:279:42: DCO023 "relation_name" argument should be described in the docstring, more information: https://github.com/jdkandersson/flake8-docstrings-complete#fix-dco023
./lib/charms/nginx_ingress_integrator/v0/ingress.py:347:5: DCO061 a function/ method without attributes should not have the attributes section in the docstring, more information: https://github.com/jdkandersson/flake8-docstrings-complete#fix-dco061
./lib/charms/nginx_ingress_integrator/v0/ingress.py:386:5: DCO061 a function/ method without attributes should not have the attributes section in the docstring, more information: https://github.com/jdkandersson/flake8-docstrings-complete#fix-dco061
./lib/charms/nginx_ingress_integrator/v0/nginx_route.py:294:9: DCO050 a function/ method that raises an exception should have the raises section in the docstring, more information: https://github.com/jdkandersson/flake8-docstrings-complete#fix-dco050
./lib/charms/nginx_ingress_integrator/v0/nginx_route.py:363:5: DCO050 a function/ method that raises an exception should have the raises section in the docstring, more information: https://github.com/jdkandersson/flake8-docstrings-complete#fix-dco050
./src/charm.py:72:9: DCO020 a function/ method with arguments should have the arguments section in the docstring, more information: https://github.com/jdkandersson/flake8-docstrings-complete#fix-dco020
./src/charm.py:86:9: DCO020 a function/ method with arguments should have the arguments section in the docstring, more information: https://github.com/jdkandersson/flake8-docstrings-complete#fix-dco020
./src/charm.py:91:13: DCO030 function/ method that returns a value should have the returns section in the docstring, more information: https://github.com/jdkandersson/flake8-docstrings-complete#fix-dco030
./src/charm.py:93:13: DCO030 function/ method that returns a value should have the returns section in the docstring, more information: https://github.com/jdkandersson/flake8-docstrings-complete#fix-dco030
./src/charm.py:95:9: DCO030 function/ method that returns a value should have the returns section in the docstring, more information: https://github.com/jdkandersson/flake8-docstrings-complete#fix-dco030
./src/charm.py:98:9: DCO020 a function/ method with arguments should have the arguments section in the docstring, more information: https://github.com/jdkandersson/flake8-docstrings-complete#fix-dco020
./src/charm.py:106:25: DCO030 function/ method that returns a value should have the returns section in the docstring, more information: https://github.com/jdkandersson/flake8-docstrings-complete#fix-dco030
./src/charm.py:108:25: DCO030 function/ method that returns a value should have the returns section in the docstring, more information: https://github.com/jdkandersson/flake8-docstrings-complete#fix-dco030
./src/charm.py:110:21: DCO030 function/ method that returns a value should have the returns section in the docstring, more information: https://github.com/jdkandersson/flake8-docstrings-complete#fix-dco030
./src/charm.py:113:17: DCO030 function/ method that returns a value should have the returns section in the docstring, more information: https://github.com/jdkandersson/flake8-docstrings-complete#fix-dco030
./src/charm.py:114:9: DCO030 function/ method that returns a value should have the returns section in the docstring, more information: https://github.com/jdkandersson/flake8-docstrings-complete#fix-dco030
./src/charm.py:117:9: DCO020 a function/ method with arguments should have the arguments section in the docstring, more information: https://github.com/jdkandersson/flake8-docstrings-complete#fix-dco020
./src/charm.py:120:13: DCO030 function/ method that returns a value should have the returns section in the docstring, more information: https://github.com/jdkandersson/flake8-docstrings-complete#fix-dco030
./src/charm.py:124:13: DCO030 function/ method that returns a value should have the returns section in the docstring, more information: https://github.com/jdkandersson/flake8-docstrings-complete#fix-dco030
./src/charm.py:126:9: DCO030 function/ method that returns a value should have the returns section in the docstring, more information: https://github.com/jdkandersson/flake8-docstrings-complete#fix-dco030
./src/charm.py:129:9: DCO020 a function/ method with arguments should have the arguments section in the docstring, more information: https://github.com/jdkandersson/flake8-docstrings-complete#fix-dco020
./src/charm.py:132:13: DCO030 function/ method that returns a value should have the returns section in the docstring, more information: https://github.com/jdkandersson/flake8-docstrings-complete#fix-dco030
./src/charm.py:136:13: DCO030 function/ method that returns a value should have the returns section in the docstring, more information: https://github.com/jdkandersson/flake8-docstrings-complete#fix-dco030
./src/charm.py:138:9: DCO030 function/ method that returns a value should have the returns section in the docstring, more information: https://github.com/jdkandersson/flake8-docstrings-complete#fix-dco030
./src/charm.py:310:9: DCO030 function/ method that returns a value should have the returns section in the docstring, more information: https://github.com/jdkandersson/flake8-docstrings-complete#fix-dco030
./src/charm.py:404:9: DCO030 function/ method that returns a value should have the returns section in the docstring, more information: https://github.com/jdkandersson/flake8-docstrings-complete#fix-dco030
./src/charm.py:415:5: DCO060 a class with attributes should have the attributes section in the docstring, more information: https://github.com/jdkandersson/flake8-docstrings-complete#fix-dco060
./src/charm.py:493:9: DCO020 a function/ method with arguments should have the arguments section in the docstring, more information: https://github.com/jdkandersson/flake8-docstrings-complete#fix-dco020
./src/charm.py:510:9: DCO030 function/ method that returns a value should have the returns section in the docstring, more information: https://github.com/jdkandersson/flake8-docstrings-complete#fix-dco030
./src/charm.py:515:9: DCO030 function/ method that returns a value should have the returns section in the docstring, more information: https://github.com/jdkandersson/flake8-docstrings-complete#fix-dco030
./src/charm.py:524:9: DCO030 function/ method that returns a value should have the returns section in the docstring, more information: https://github.com/jdkandersson/flake8-docstrings-complete#fix-dco030
./src/charm.py:546:9: DCO030 function/ method that returns a value should have the returns section in the docstring, more information: https://github.com/jdkandersson/flake8-docstrings-complete#fix-dco030
./src/charm.py:549:9: DCO020 a function/ method with arguments should have the arguments section in the docstring, more information: https://github.com/jdkandersson/flake8-docstrings-complete#fix-dco020
./src/charm.py:552:9: DCO030 function/ method that returns a value should have the returns section in the docstring, more information: https://github.com/jdkandersson/flake8-docstrings-complete#fix-dco030
./src/charm.py:555:9: DCO024 "current_svc_hostnames" argument should not be described in the docstring, more information: https://github.com/jdkandersson/flake8-docstrings-complete#fix-dco024
./src/charm.py:598:9: DCO020 a function/ method with arguments should have the arguments section in the docstring, more information: https://github.com/jdkandersson/flake8-docstrings-complete#fix-dco020
./src/charm.py:627:9: DCO020 a function/ method with arguments should have the arguments section in the docstring, more information: https://github.com/jdkandersson/flake8-docstrings-complete#fix-dco020
./src/charm.py:644:9: DCO020 a function/ method with arguments should have the arguments section in the docstring, more information: https://github.com/jdkandersson/flake8-docstrings-complete#fix-dco020
./src/charm.py:697:9: DCO020 a function/ method with arguments should have the arguments section in the docstring, more information: https://github.com/jdkandersson/flake8-docstrings-complete#fix-dco020
./src/charm.py:742:9: DCO020 a function/ method with arguments should have the arguments section in the docstring, more information: https://github.com/jdkandersson/flake8-docstrings-complete#fix-dco020
./src/charm.py:742:9: DCO050 a function/ method that raises an exception should have the raises section in the docstring, more information: https://github.com/jdkandersson/flake8-docstrings-complete#fix-dco050
./src/charm.py:807:9: DCO030 function/ method that returns a value should have the returns section in the docstring, more information: https://github.com/jdkandersson/flake8-docstrings-complete#fix-dco030
./src/charm.py:810:9: DCO020 a function/ method with arguments should have the arguments section in the docstring, more information: https://github.com/jdkandersson/flake8-docstrings-complete#fix-dco020
./src/charm.py:812:9: DCO030 function/ method that returns a value should have the returns section in the docstring, more information: https://github.com/jdkandersson/flake8-docstrings-complete#fix-dco030
./src/charm.py:817:9: DCO020 a function/ method with arguments should have the arguments section in the docstring, more information: https://github.com/jdkandersson/flake8-docstrings-complete#fix-dco020
./src/charm.py:835:9: DCO030 function/ method that returns a value should have the returns section in the docstring, more information: https://github.com/jdkandersson/flake8-docstrings-complete#fix-dco030
./src/charm.py:847:9: DCO020 a function/ method with arguments should have the arguments section in the docstring, more information: https://github.com/jdkandersson/flake8-docstrings-complete#fix-dco020
./src/charm.py:875:9: DCO020 a function/ method with arguments should have the arguments section in the docstring, more information: https://github.com/jdkandersson/flake8-docstrings-complete#fix-dco020
./src/charm.py:901:9: DCO050 a function/ method that raises an exception should have the raises section in the docstring, more information: https://github.com/jdkandersson/flake8-docstrings-complete#fix-dco050
./src/charm.py:901:9: DCO055 a function/ method that re-raises exceptions should describe at least one exception in the raises section of the docstring, more information: https://github.com/jdkandersson/flake8-docstrings-complete#fix-dco055
./src/charm.py:956:9: DCO020 a function/ method with arguments should have the arguments section in the docstring, more information: https://github.com/jdkandersson/flake8-docstrings-complete#fix-dco020
./src/charm.py:956:9: DCO050 a function/ method that raises an exception should have the raises section in the docstring, more information: https://github.com/jdkandersson/flake8-docstrings-complete#fix-dco050
./src/charm.py:956:9: DCO055 a function/ method that re-raises exceptions should describe at least one exception in the raises section of the docstring, more information: https://github.com/jdkandersson/flake8-docstrings-complete#fix-dco055
./src/helpers.py:24:9: DCO030 function/ method that returns a value should have the returns section in the docstring, more information: https://github.com/jdkandersson/flake8-docstrings-complete#fix-dco030
./src/helpers.py:25:5: DCO030 function/ method that returns a value should have the returns section in the docstring, more information: https://github.com/jdkandersson/flake8-docstrings-complete#fix-dco030
./tests/integration/conftest.py:51:5: DCO010 docstring should be defined for a function/ method/ class, more information: https://github.com/jdkandersson/flake8-docstrings-complete#fix-dco010
./tests/integration/conftest.py:66:5: DCO010 docstring should be defined for a function/ method/ class, more information: https://github.com/jdkandersson/flake8-docstrings-complete#fix-dco010
./tests/integration/conftest.py:88:5: DCO010 docstring should be defined for a function/ method/ class, more information: https://github.com/jdkandersson/flake8-docstrings-complete#fix-dco010
./tests/integration/conftest.py:101:5: DCO010 docstring should be defined for a function/ method/ class, more information: https://github.com/jdkandersson/flake8-docstrings-complete#fix-dco010
./tests/integration/conftest.py:115:5: DCO010 docstring should be defined for a function/ method/ class, more information: https://github.com/jdkandersson/flake8-docstrings-complete#fix-dco010
./tests/integration/conftest.py:131:5: DCO010 docstring should be defined for a function/ method/ class, more information: https://github.com/jdkandersson/flake8-docstrings-complete#fix-dco010
./tests/integration/test_nginx_route.py:28:5: DCO020 a function/ method with arguments should have the arguments section in the docstring, more information: https://github.com/jdkandersson/flake8-docstrings-complete#fix-dco020
./tests/integration/test_nginx_route.py:46:5: DCO030 function/ method that returns a value should have the returns section in the docstring, more information: https://github.com/jdkandersson/flake8-docstrings-complete#fix-dco030
./tests/integration/test_nginx_route.py:50:5: DCO020 a function/ method with arguments should have the arguments section in the docstring, more information: https://github.com/jdkandersson/flake8-docstrings-complete#fix-dco020
./tests/integration/test_nginx_route.py:55:13: DCO030 function/ method that returns a value should have the returns section in the docstring, more information: https://github.com/jdkandersson/flake8-docstrings-complete#fix-dco030
./tests/integration/test_relation.py:64:5: DCO010 docstring should be defined for a function/ method/ class, more information: https://github.com/jdkandersson/flake8-docstrings-complete#fix-dco010
./tests/integration/test_relation.py:99:5: DCO010 docstring should be defined for a function/ method/ class, more information: https://github.com/jdkandersson/flake8-docstrings-complete#fix-dco010
./tests/integration/test_relation.py:155:5: DCO010 docstring should be defined for a function/ method/ class, more information: https://github.com/jdkandersson/flake8-docstrings-complete#fix-dco010
./tests/unit/test_charm.py:18:1: DCO010 docstring should be defined for a function/ method/ class, more information: https://github.com/jdkandersson/flake8-docstrings-complete#fix-dco010
./tests/unit/test_charm.py:1247:1: DCO010 docstring should be defined for a function/ method/ class, more information: https://github.com/jdkandersson/flake8-docstrings-complete#fix-dco010
./tests/unit/test_charm.py:1253:1: DCO010 docstring should be defined for a function/ method/ class, more information: https://github.com/jdkandersson/flake8-docstrings-complete#fix-dco010
./tests/unit/test_charm.py:1254:5: DCO010 docstring should be defined for a function/ method/ class, more information: https://github.com/jdkandersson/flake8-docstrings-complete#fix-dco010
./tests/unit/test_charm.py:1300:1: DCO010 docstring should be defined for a function/ method/ class, more information: https://github.com/jdkandersson/flake8-docstrings-complete#fix-dco010
./tests/unit/test_charm.py:1307:5: DCO010 docstring should be defined for a function/ method/ class, more information: https://github.com/jdkandersson/flake8-docstrings-complete#fix-dco010
./tests/unit/test_charm.py:1729:9: DCO010 docstring should be defined for a function/ method/ class, more information: https://github.com/jdkandersson/flake8-docstrings-complete#fix-dco010
./tests/unit/test_charm.py:2079:1: DCO010 docstring should be defined for a function/ method/ class, more information: https://github.com/jdkandersson/flake8-docstrings-complete#fix-dco010
./tests/unit/test_library_ingress.py:19:1: DCO010 docstring should be defined for a function/ method/ class, more information: https://github.com/jdkandersson/flake8-docstrings-complete#fix-dco010
./tests/unit/test_library_ingress.py:20:5: DCO010 docstring should be defined for a function/ method/ class, more information: https://github.com/jdkandersson/flake8-docstrings-complete#fix-dco010
./tests/unit/test_library_ingress.py:39:9: DCO010 docstring should be defined for a function/ method/ class, more information: https://github.com/jdkandersson/flake8-docstrings-complete#fix-dco010
./tests/unit/test_library_ingress.py:40:13: DCO010 docstring should be defined for a function/ method/ class, more information: https://github.com/jdkandersson/flake8-docstrings-complete#fix-dco010
./tests/unit/test_library_ingress.py:60:9: DCO010 docstring should be defined for a function/ method/ class, more information: https://github.com/jdkandersson/flake8-docstrings-complete#fix-dco010
./tests/unit/test_library_ingress.py:61:13: DCO010 docstring should be defined for a function/ method/ class, more information: https://github.com/jdkandersson/flake8-docstrings-complete#fix-dco010
./tests/unit/test_library_ingress.py:91:9: DCO010 docstring should be defined for a function/ method/ class, more information: https://github.com/jdkandersson/flake8-docstrings-complete#fix-dco010
./tests/unit/test_library_ingress.py:92:13: DCO010 docstring should be defined for a function/ method/ class, more information: https://github.com/jdkandersson/flake8-docstrings-complete#fix-dco010
./tests/unit/test_library_ingress.py:130:9: DCO010 docstring should be defined for a function/ method/ class, more information: https://github.com/jdkandersson/flake8-docstrings-complete#fix-dco010
./tests/unit/test_library_ingress.py:131:13: DCO010 docstring should be defined for a function/ method/ class, more information: https://github.com/jdkandersson/flake8-docstrings-complete#fix-dco010
./tests/unit/test_library_ingress.py:144:1: DCO010 docstring should be defined for a function/ method/ class, more information: https://github.com/jdkandersson/flake8-docstrings-complete#fix-dco010
./tests/unit/test_library_ingress.py:145:5: DCO010 docstring should be defined for a function/ method/ class, more information: https://github.com/jdkandersson/flake8-docstrings-complete#fix-dco010
```